### PR TITLE
docs: update bash commands from npm to pnpm equivalent , add discord dev mode description

### LINKED ISSUE #559: suggestions in improving the contributing.md docs
In my earlier attempt in fixing one of the issues, i faced the following challenges which i describe below as a form of updating the contributing.md documentation 

line 25:
  At this point make sure the Developer Mode is enabled in your user settings of Discord
  ```mermaid
  graph LR 
  A(Discord Application) --> B(USER SETTINGS)--> C(APP SETTINGS)--> D(Adanced)--> E(Developer Mode = Enable) 
```
alternatively we could say this at the beginning of the page on line 3  :
`To develop this project locally, you will need to be running [NodeJS v20](https://nodejs.org/en/) , [pnpm v8](https://pnpm.io/). and have enabled Developer Mode in Discord` 

line 40 could be updated to `pnpm i --frozen-lockfile`  or `pnpm install`
`npm ci` refused to work for me showing the following as error
`The npm ci command can only install with an existing package-lock.json or
npm ERR! npm-shrinkwrap.json with lockfileVersion >= 1. Run an install with npm@5 or
npm ERR! later to generate a package-lock.json file, then try again.`

line 46 could be updated to `pnpm run build`.
line 73 could be updated to `pnpm start`.
line 83 could be updated to `pnpm run lint`.

**Desktop (please complete the following information):**
 - OS: [MacOS 13.4.1]
 - Browser: [Chrome]
 - Version: [122.0.6261.94 

**Additional context**
I am happy to work on this issue once the correct approach and wording is decided.

### PR DESCRIPTION
**Pull request by RGHANILOO**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #559

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ Next, head over to [MongoDB Atlas](https://www.mongodb.com/cloud/atlas) and eith
 
 The bot uses two Discord webhooks, one for debug logging and one for moderation action logs. For local testing, you can use the same webhook URL for both. Right-click on a channel in your server, select `Edit Channel`, and pick the `Integrations` tab on the sidebar. Click the `Create Webhook` button, give it a name, and copy the URL. Paste that as `DEBUG_HOOK` and `MOD_HOOK` in your `.env` file.
 
+If you haven't already enabled Developer Mode in your Discord client, You can do this by opening your user/app setting and selecting `Advanced`,  then toggling `Developer Mode` on.
```
Comment (naomi-lgbt) [inline CONTRIBUTING.md]:
```suggestion
If you haven't already enabled Developer Mode in your Discord client, you can do this by opening your user/app setting and selecting `Advanced`,  then toggling `Developer Mode` on.
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
One nit

File: CONTRIBUTING.md (hunk 1 above)
Comment (RGHANILOO) [reply CONTRIBUTING.md]: resolved in commit #91df32b

**Comment by RGHANILOO** [comment]
> One nit

Seen to. 😃

**Review by naomi-lgbt — APPROVED** [review]
Right on! Thanks for taking care of this 💜

**Pull request merged**